crates/bevy_ecs/src/query/filter.rs

```diff
@@ -588,52 +574,81 @@ macro_rules! impl_tick_filter {
                         self.table_ticks = Some(table.get_column(state.component_id).unwrap().get_ticks_slice().into());
                     }
-                    StorageType::SparseSet => self.entities = Some(archetype.entities().into()),
+                    StorageType::SparseSet => {},
                 }
             }
 
-            unsafe fn table_fetch(&mut self, table_row: usize) -> bool {
-                $is_detected(&*(self.table_ticks.unwrap_or_else(|| debug_checked_unreachable()).get(table_row)).deref(), self.last_change_tick, self.change_tick)
-            }
-
-            unsafe fn archetype_fetch(&mut self, archetype_index: usize) -> bool {
+            #[inline(always)]
+            unsafe fn fetch(&mut self, entity: &Entity, table_row: &usize) -> Self::Item {
                 match T::Storage::STORAGE_TYPE {
                     StorageType::Table => {
-                        let table_row = *self.entity_table_rows.unwrap_or_else(|| debug_checked_unreachable()).get(archetype_index);
-                        $is_detected(&*(self.table_ticks.unwrap_or_else(|| debug_checked_unreachable()).get(table_row)).deref(), self.last_change_tick, self.change_tick)
+                        $is_detected(&*(
+                            self.table_ticks
+                                .unwrap_or_else(|| debug_checked_unreachable())
+                                .get(*table_row))
+                                .deref(),
+                            self.last_change_tick,
+                            self.change_tick
+                        )
                     }
                     StorageType::SparseSet => {
-                        let entity = *self.entities.unwrap_or_else(|| debug_checked_unreachable()).get(archetype_index);
                         let ticks = self
                             .sparse_set
                             .unwrap_or_else(|| debug_checked_unreachable())
-                            .get_ticks(entity)
+                            .get_ticks(*entity)
                             .map(|ticks| &*ticks.get())
                             .cloned()
-                            .unwrap();
+                            .unwrap_or_else(|| debug_checked_unreachable());
                         $is_detected(&ticks, self.last_change_tick, self.change_tick)
                     }
                 }
             }
 
-            #[inline]
-            unsafe fn table_filter_fetch(&mut self, table_row: usize) -> bool {
-                self.table_fetch(table_row)
-            }
-
-            #[inline]
-            unsafe fn archetype_filter_fetch(&mut self, archetype_index: usize) -> bool {
-                self.archetype_fetch(archetype_index)
+            #[inline(always)]
+            unsafe fn filter_fetch(&mut self, _entity: &Entity, _table_row: &usize) -> Self::Item {
+                // Explicitly repeated code here to avoid potential poor inlining
```

Comment:
was this benchmarked or are we massively regressing code quality wise for potentially no reason